tests/test_georaster.py

```diff
@@ -26,6 +26,46 @@
 DO_PLOT = False
 
 
+def run_gdal_proximity(
+    input_raster: gu.Raster, target_values: list[float] | None, distunits: str = "GEO"
+) -> np.ndarray:
+    """Run GDAL's ComputeProximity and return the read numpy array."""
+    # Rasterio strongly recommends against importing gdal along rio, so this is done here instead.
+    from osgeo import gdal, gdalconst
+
+    # Initiate empty GDAL raster for proximity output
+    drv = gdal.GetDriverByName("MEM")
+    proxy_ds = drv.Create("", input_raster.shape[1], input_raster.shape[0], 1, gdal.GetDataTypeByName("Float32"))
+    proxy_ds.GetRasterBand(1).SetNoDataValue(-9999)
+
+    # Save input in temporary file to read with GDAL
+    # (avoids the nightmare of setting nodata, transform, crs in GDAL format...)
+    temp_dir = tempfile.TemporaryDirectory()
```

Comment:
I've realized the hard way that this does not delete the directory if the program crashes before this goes out of scope. Please change this to:
```python
with tempfile.TemporaryDirectory() as temp_dir:
    temp_path = os.path.join(temp_dir, "input.tif")  # The temp_dir variable is now just a string of the path.
    ...
```

Reply:
Done!